src/js/reducers/inspector.js

```diff
@@ -0,0 +1,38 @@
+/* eslint new-cap:0 */
+'use strict';
+
+var Immutable = require('immutable');
+
+var actions = require('../constants/actions');
+var getIn = require('../util/immutable-utils').getIn;
+var setIn = require('../util/immutable-utils').setIn;
+
+function inspectorReducer(state, action) {
+  if (typeof state === 'undefined') {
+    return Immutable.fromJS({
+      selected: null,
+      expandedLayers: {}
+    });
+  }
+
+  if (action.type === actions.SELECT_MARK) {
+    return state.set('selected', action.markId);
+  }
+
+  if (action.type === actions.EXPAND_LAYERS) {
+    return action.layerIds.reduce(function(newState, layerId) {
+      return setIn(newState, 'expandedLayers.' + layerId, true);
+    }, state);
+  }
```

Comment:
@deathbearbrown thank you for the run-down on `.setIn`, with that API and the wrapper I wrote for it this reducer finally looks like a reducer rather than a clumsy merge-r!